tensorflow-examples/src/main/java/org/tensorflow/model/examples/cnn/alexnet/AlexNetModel.java

```diff
@@ -0,0 +1,294 @@
+
+        //Layer 2
+        Relu<TFloat32> relu2 = alexNetConv2DLayer("2", tf, norm1, new int[]{5, 5, 96, 256}, 256);
+        MaxPool<TFloat32> pool2 = alexNetMaxPool(tf, relu2);
+        LocalResponseNormalization<TFloat32> norm2 = alexNetModelLRN(tf, pool2);
+
+        //Layer 3
+        Relu<TFloat32> relu3 = alexNetConv2DLayer("3", tf, norm2, new int[]{3, 3, 256, 384}, 384);
+        LocalResponseNormalization<TFloat32> norm3 = alexNetModelLRN(tf, relu3);
+
+        //Layer 4
+        Relu<TFloat32> relu4 = alexNetConv2DLayer("4", tf, norm3, new int[]{3, 3, 384, 384}, 384);
+
+        //Layer 5
+        Relu<TFloat32> relu5 = alexNetConv2DLayer("2", tf, relu4, new int[]{3, 3, 384, 256}, 256);
+        MaxPool<TFloat32> pool5 = alexNetMaxPool(tf, relu5);
+        LocalResponseNormalization<TFloat32> norm5 = alexNetModelLRN(tf, pool5);
+
+        Reshape<TFloat32> flatten = alexNetFlatten(tf, pool5);
+
+        Add<TFloat32> loss = buildFCLayersAndRegularization(tf, labels, flatten);
+
+        Optimizer optimizer = new Adam(graph, 0.001f, 0.9f, 0.999f, 1e-8f);
+
+        optimizer.minimize(loss, TRAIN);
+
+        tf.init();
+
+        return graph;
+    }
+
+    public static Add<TFloat32> buildFCLayersAndRegularization(Ops tf, Placeholder<TUint8> labels, Reshape<TFloat32> flatten) {
+        int fcBiasShape = 500;
+        int[] fcWeightShape = {4096, fcBiasShape};
+
+        Variable<TFloat32> fc1Weights = tf.variable(tf.math.mul(tf.random
+                .truncatedNormal(tf.array(fcWeightShape), TFloat32.DTYPE,
+                        TruncatedNormal.seed(SEED)), tf.constant(0.1f)));
+        Variable<TFloat32> fc1Biases = tf
+                .variable(tf.fill(tf.array(new int[]{fcBiasShape}), tf.constant(0.1f)));
+        Relu<TFloat32> fcRelu = tf.nn
+                .relu(tf.math.add(tf.linalg.matMul(flatten, fc1Weights), fc1Biases));
+
+        // Softmax layer
+        Variable<TFloat32> fc2Weights = tf.variable(tf.math.mul(tf.random
+                .truncatedNormal(tf.array(fcBiasShape, NUM_LABELS), TFloat32.DTYPE,
+                        TruncatedNormal.seed(SEED)), tf.constant(0.1f)));
+        Variable<TFloat32> fc2Biases = tf
+                .variable(tf.fill(tf.array(new int[]{NUM_LABELS}), tf.constant(0.1f)));
+
+        Add<TFloat32> logits = tf.math.add(tf.linalg.matMul(fcRelu, fc2Weights), fc2Biases);
+
+        // Predicted outputs
+        tf.withName(OUTPUT_NAME).nn.softmax(logits);
+
+        // Loss function & regularization
+        OneHot<TFloat32> oneHot = tf
+                .oneHot(labels, tf.constant(NUM_LABELS), tf.constant(1.0f), tf.constant(0.0f));
+        SoftmaxCrossEntropyWithLogits<TFloat32> batchLoss = tf.nn.raw
+                .softmaxCrossEntropyWithLogits(logits, oneHot);
+        Mean<TFloat32> labelLoss = tf.math.mean(batchLoss.loss(), tf.constant(0));
+        Add<TFloat32> regularizers = tf.math.add(tf.nn.l2Loss(fc1Weights), tf.math
```

Comment:
Let's think, do we need regularization here, maybe better to remove it and refactor dense layers in separate procedures